miner/pool.go

```diff
@@ -0,0 +1,48 @@
+package miner
+
+import (
+	"reflect"
+	"sync"
+)
+
+func NewMemPool(tp reflect.Type) *MemPool {
+	return &MemPool{sync.RWMutex{}, tp, make(map[interface{}]interface{})}
+}
+
+type MemPool struct {
+	mu    sync.RWMutex
+	t     reflect.Type
```

Comment:
i think this is redundant since you are only using interface types as return values